src/main/printutils.c

```diff
@@ -413,7 +413,7 @@ int Rstrwid(const char *str, int slen, cetype_t ienc, int quote)
 	    unsigned int k; /* not wint_t as it might be signed */
 	    wchar_t wc;
 	    int res = useUTF8 ? (int) utf8toucs(&wc, p):
-		(int) mbrtowc(&wc, p, R_MB_CUR_MAX, NULL);
+		(int) mbrtowc(&wc, p, R_MB_CUR_MAX, &mb_st);
```

Comment:
This seems clearly like an oversight, `mb_st` is initialized on line 411, but then unused.  This is also the source of the original bug.

To test:

    LC_CTYPE=en_US.utf8 R
    x <- "\xe9"
    Encoding(x) <- "latin1"
    x